sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java

```diff
@@ -39,8 +39,9 @@ public OfferEvaluator(StateStore stateStore, OfferRequirementProvider offerRequi
 
     public List<OfferRecommendation> evaluate(OfferRequirement offerRequirement, List<Offer> offers)
             throws StateStoreException, InvalidRequirementException {
+        List<OfferEvaluationStage> evaluationStages = getEvaluationPipeline(offerRequirement);
+
         for (int i = 0; i < offers.size(); ++i) {
-            List<OfferEvaluationStage> evaluationStages = getEvaluationPipeline(offerRequirement);
```

Comment:
The evaluation stages should be recreated for every offer as they mutate their state.